File: public/examples/integrations/toolkits/microsoft/outlook_mail/update_draft_email_example_call_tool.py
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "you@example.com"
TOOL_NAME = "Microsoft.UpdateDraftEmail"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    "message_id": "1234567890",
    "subject": "My new subject",
    "body": "My new body",
    "to_add": ["johndoe@example.com"],
    "to_remove": ["a@example.com"],
    "cc_add": ["b@example.com"],
    "cc_remove": ["c@example.com"],
    "bcc_add": ["d@example.com"],
    "bcc_remove": ["e@example.com"],
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(response)

File: public/examples/integrations/toolkits/microsoft/outlook_mail/update_draft_email_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "you@example.com";
const TOOL_NAME = "Microsoft.UpdateDraftEmail";

// Start the authorization process
const authResponse = await client.tools.authorize({
  tool_name: TOOL_NAME,
  user_id: USER_ID,
});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  message_id: "1234567890",
  subject: "My new subject",
  body: "My new body",
  to_add: ["johndoe@example.com"],
  to_remove: ["a@example.com"],
  cc_add: ["b@example.com"],
  cc_remove: ["c@example.com"],
  bcc_add: ["d@example.com"],
  bcc_remove: ["e@example.com"],
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(response);
